storage_handler.go

```diff
@@ -86,39 +108,13 @@ func (h *StorageHandler) handleGET(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	s3key := h.s3KeyForFileID(sf)
-
-	objectInput := s3.GetObjectInput{
-		Bucket: &h.S3Bucket,
-		Key:    &s3key,
-	}
-
-	resp, err := h.S3API.GetObjectWithContext(r.Context(), &objectInput)
+	presignedURL, err := h.Storage.GetObjectPresignedURL(r.Context(), h.keyForFileID(sf))
```

Comment:
Once the user gets the presigned URL, that's up to OSN. (I've confirmed that once the URL expires, OSN responds with a 403 Forbidden.)

That being said, _trusted_ users can simply try the original URL again to get a new presigned URL.

I suspect most people won't keep the presigned URL, since clients should handle the redirect transparently, but we may want to make that clear in the docs.